camelConnector/src/main/java/io/vantiq/extsrc/camelconn/connector/CamelHandleConfiguration.java

```diff
@@ -258,12 +250,43 @@ public boolean runCamelApp(Map<String, Object> camelConfig, Map<String, Object>
             // This is checked in the caller
             //noinspection unchecked
             Map<String, Object> input = (Map<String, Object>) camelConfig.get(PROPERTY_VALUES);
+            //noinspection unchecked
+            Map<String, List<String>> raw = (Map<String, List<String>>) camelConfig.get(RAW_REQUIRED);
+            List<String> rawRequired = new ArrayList<>();
+            // Create a list from the various categories that may be in this set...
+            raw.forEach( (k, v) -> {
+                rawRequired.addAll(v);
+            });
             // Camel wants these as a Java Properties object, so perform the conversion as required.
             Properties propVals = null;
             if (input != null) {
                 propVals = new Properties(input.size());
                 for (Map.Entry<String, Object> p : input.entrySet()) {
-                    propVals.setProperty(p.getKey(), p.getValue().toString());
+                    if (p.getValue() != null) {
+                        // Assembly mechanism may send nulls for empty values.  Ignore them here.
+                        if (rawRequired.contains(p.getKey()) && p.getValue() instanceof String) {
+                            String val = (String) p.getValue();
+                            // If the caller has already wrapped our values with RAW()/RAW{}, we needn't bother.
+                            boolean suppress = (val.startsWith("RAW(") && val.endsWith(")")) ||
+                                    (val.startsWith("RAW{") && val.endsWith("}"));
+                            if (!suppress || !(val.contains(")") && val.contains("}"))) {
+                                String startRaw = "RAW(";
+                                String endRaw = ")";
+                                if (val.contains(")")) {
+                                    if (!val.contains("}")) {
+                                        startRaw = "RAW{";
+                                        endRaw = "}";
+                                    }
+                                }
+                                propVals.setProperty(p.getKey(), startRaw + val + endRaw);
+                            } else {
+                                // If we're suppressing or we havwe a value with both ) & }, leave things alone
```

Comment:
```suggestion
                                // If we're suppressing or we have a value with both ) & }, leave things alone
```

Reply:
Why does it not need to wrap if the string has both a `)` and a `}`? If it's just that it would mess with the RAW parsing, that seems like it should cause an error rather than just ignore it.

Reply:
I think that the receiver would see an error (since the property value isn’t wrapped), but I don’t know that for certain. So, rather than our refusing to do the work, it seemed better to pass it thru and let the underlying systems sort things out. In this case, it is our generation that decided that things likely needed to be wrapped; it may be that that’s not required — it may be that the underlying systems can sort it out. Since this is 5th party integration atop 4th party atop 3rd party, much of it open source, it seemed better here to just leave things in their natural state and let the _nature_ take its course.

Reply:
(Sorry — didn’t see this remark yesterday, so I’m a bit late with the response.)

Reply:
Updated code -- streamlined a bit and added comments about what & why.  Also added ability to bypass auto wrapping.  Added tests